### PR TITLE
F/us 18 indoor directions display

### PR DESCRIPTION
**Pull request by aiceball**
**Context**

This PR delivers displaying indoor directions using google Polyline approach.

It also brings a couple of small fixes to the application:
1. Some adjacency matrix corrections for proper pathing
2. Outdoor route selection fix (#143 )
3. Overlay error correction

**Example**
Route from H801 to H 903 using the stairs:

Floor 8 directions
image

Floor 9 directions
image

**Test coverage**
image

### CODE REVIEW

File: src/app/core/models/indoor-route.ts [hunk 1]
```diff
@@ -1,17 +1,28 @@
...
+        Object.assign({}, indoorPoiToCoordinates, indoorWalkingPathCoordinates);
+
     constructor(startLocation: string, endLocation: string, disability: boolean, routeSteps: RouteStep[], distance: number) {
-        // TODO fix this
-        const moduleKey = 'default';
-        const coords: {[coordName: string]: Coordinates} = indoorPoiCoordinates[moduleKey];
-        this.startCoordinates = coords[startLocation];
-        this.endCoordinates = coords[endLocation];
+        /// TODO move this to a function
+        this.startCoordinates = new Coordinates(
```
Comment (ClaudiaLapalme) [inline src/app/core/models/indoor-route.ts:16]: this `TODO` should probably be addressed in this PR

File: src/app/core/services/map.service.ts [hunk 2]
```diff
@@ -208,31 +211,95 @@ export class MapService {
...
+    }
+
+    /**
+     * Filter method allowing us to display only the route passed from the routes page
+     * @param res result returned by the google API
+     * @param startTime expected true start time
+     * @param endTime expected true end time
+     * @returns index of the route matching start and end times
+     */
+    private filterOutRoute(res: google.maps.DirectionsResult, startTime: Date, endTime: Date): number {
```
Comment (ClaudiaLapalme) [inline src/app/core/services/map.service.ts:255]: Is this related to #143?

File: src/app/core/services/map.service.ts [hunk 3]
```diff
@@ -1,18 +1,22 @@
 import { ElementRef, Injectable } from '@angular/core';
```
Comment (ClaudiaLapalme) [inline src/app/core/services/map.service.ts:1]: Check the `index.html` in the coverage folder of the project to see which of the new functions are lacking in testing. There are a few more tests that could be added to improve code coverage

File: src/app/core/services/map.service.ts [hunk 4]
```diff
@@ -208,31 +211,95 @@ export class MapService {
...
+        return -1;
+    }
+
+    /**
+     * Method to display indoor routing using google Polylines. Tracks previous drawing, clearing it out per request
+     * @param map map object to render the route on
+     * @param indoorRoute route to render on the map
+     * @param indoorMapLevel floor for which to draw the polyline
+     */
+    displayIndoorRoute(map: google.maps.Map, indoorRoute: IndoorRoute, indoorMapLevel: number) {
```
Comment (ClaudiaLapalme) [inline src/app/core/services/map.service.ts:271]: Missing return type

File: src/app/core/services/map.service.ts [hunk 5]
```diff
@@ -208,31 +211,95 @@ export class MapService {
...
+        // Clear polylines previously drawn in order to keep the needed polylines per floor only
+        if (this.drawnPolyline != null) {
+            this.drawnPolyline.setMap(null);
+        }
+
+        // If there are route steps for current floor, draw a directions polyline for that floor
+        indoorRoute.routeSteps.forEach(routeStep => {
+            if (routeStep.startCoordinate.getFloorNumber() === indoorMapLevel
+                && routeStep.endCoordinate.getFloorNumber() === indoorMapLevel) {
+                this.drawnPolyline = this.googleApis.createPolyline(mapCoordinatesArrayToLatLng(routeStep.path), true, 'red', 1.0, 2);
```
Comment (ClaudiaLapalme) [inline src/app/core/services/map.service.ts:285]: Should we change the polyline colour of the shuttle to match the red of the indoor routes? It will provide uniformity and black is going to be confusing on the indoor maps.

File: src/app/rendered-routes/rendered-routes.page.ts [hunk 6]
```diff
@@ -2,12 +2,13 @@ import {
     AfterViewInit,
```
Comment (ClaudiaLapalme) [inline src/app/rendered-routes/rendered-routes.page.ts:2]: this page could benefit from having more tests

File: src/app/home/home.page.spec.ts [hunk 7]
```diff
@@ -194,6 +194,7 @@ describe('HomePage', () => {
             lng = 0;
         }
 
+        // Flaky
```
Comment (ClaudiaLapalme) [inline src/app/home/home.page.spec.ts:197]: why flaky?

**Comment by ClaudiaLapalme** [comment]
After discussion with the team members, the current levels of testing are accepted as it is difficult to tests the functions to display.

File: src/app/core/services/map.service.ts (hunk 2 above)
Comment (aiceball) [reply src/app/core/services/map.service.ts:255]: Yes, this is the fix for #143 . I will add a reference in the description

File: src/app/core/services/map.service.ts (hunk 5 above)
Comment (aiceball) [reply src/app/core/services/map.service.ts:285]: I'll defer the decision to @lara-tran , since her feature came first. Whatever is chosen, I'll apply

File: src/app/home/home.page.spec.ts (hunk 7 above)
Comment (aiceball) [reply src/app/home/home.page.spec.ts:197]: Uknown side effect, occasionally there's an async timeout that occurs. I would lean towards network instability causing the delays, hence the flakiness.

File: src/app/core/services/map.service.ts (hunk 5 above)
Comment (ClaudiaLapalme) [reply src/app/core/services/map.service.ts:285]: Black is going to be confusing as the wall outlines are black as well, though. It's also not a big deal if we keep both different colours.

Comment (lara-tran) [reply src/app/core/services/map.service.ts:285]: can you change the shuttle line to red?

File: src/app/home/home.page.spec.ts (hunk 7 above)
Comment (ClaudiaLapalme) [reply src/app/home/home.page.spec.ts:197]: In that case, your explanation here would be a better comment than just `flaky`
